gc/base/ParallelDispatcher.cpp

```diff
@@ -723,20 +723,21 @@ MM_ParallelDispatcher::expandThreadPool(MM_EnvironmentBase *env)
 	Assert_MM_false(_inShutdown);
 
 	Assert_MM_true(_threadShutdownCount == (_threadCountMaximum - 1));
-	Assert_MM_true(_threadCountMaximum == _extensions->gcThreadCount);
 
 	bool result = true;
 
-	uintptr_t supportedGCThreadCount = _extensions->configuration->supportedGCThreadCount(env);
-	uintptr_t preExpandThreadCount = _extensions->gcThreadCount;
-	uintptr_t newThreadCount = OMR_MIN(_threadCountMaximumAtCheckpointStartup, supportedGCThreadCount);
+	uintptr_t preExpandThreadCount = _threadCountMaximum;
+	uintptr_t newThreadCount = _extensions->gcThreadCount;
+
+	Assert_MM_true(newThreadCount >= preExpandThreadCount);
 
 	Trc_MM_ParallelDispatcher_expandThreadPool_params(
-			newThreadCount, _threadCountMaximumAtCheckpointStartup,
-			supportedGCThreadCount, preExpandThreadCount);
+			newThreadCount, _poolMaxCapacity,
+			_extensions->configuration->supportedGCThreadCount(env), preExpandThreadCount);
 
-	if (newThreadCount > preExpandThreadCount) {
+	result = reinitializeThreadPool(env, newThreadCount);
 
+	if ((newThreadCount > preExpandThreadCount) && result) {
```

Comment:
minor (what ever you like) but consider doing as a pair of nested ifs:
if (result = reinitialize())
    if (newThreadCount > preExpanded)
    
the most important for me was that Thread pool  was a explicit step addressed before considering that threads themselves should be addressed